pkgs/development/go-modules/generic/default.nix

```diff
@@ -37,7 +37,7 @@
 with builtins;
 
 let
-  args = removeAttrs args' [ "overrideModAttrs" "modSha256" "disabled" ];
+  args = removeAttrs args' [ "overrideModAttrs" "disabled" ];
```

Comment:
By this logic, shouldn't we expose `overrideModAttrs` as well? `disabled` seems the only one that fits the "hidden implementation detail".

Reply:
I don't think so, because `overrideModAttrs` is a hook that allows you to clobber upstream's package definition in your local overlay, not something that's used off-the-shelf in the standard NixPkgs.

The only reason you'd want `overrideModAttrs` exposed is if you were going to compose multiple overlays with the second one extending the customizations done in the first one, but at that point you might as well just merge the overlays (as indeed you have to do in Python packages).  

I'm inclined to leave it hidden for now on that rationale, but confess I don't know about the golang nix ecosystem in great depth.

Reply:
By looking at it with the question "do I need to hide this?" rather than the question "do I need to expose this?" then the rationale to me says to expose it because I don't find a compelling reason to keep it hidden.

Reply:
For reference, @basvandijk added it 11 months ago in 8d6fbef92a3684d5de4c09535b98d6e93be2f669

Reply:
Do we really should expose this to the build though? This could be part of `passthru`.

Reply:
In general I like the idea of hiding as much as possible from the builder, but in this particular case the builder already has exposure to the entire vendor directory -- so there's no leak in abstraction by having it also exposed to the hash of the vendor directory.  

Consider the case of [tflint](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/analysis/tflint/default.nix).  When we build this, it has `export GOPROXY=file:///nix/store/kybkgvsda931y2z1d39k4gy507x1ia9g-tflint-0.15.0-go-modules` set in the builder, so there's no extra dependency leakage to the builder if we also tell it that the `modSha256` of that particular directory happens to be `1qa7qfxpc43n9xyyfcd24d17g4fdcffkd57ny078ja219x13kay3`:

```
$ nix-hash --type sha256 --base32 /nix/store/kybkgvsda931y2z1d39k4gy507x1ia9g-tflint-0.15.0-go-modules
1qa7qfxpc43n9xyyfcd24d17g4fdcffkd57ny078ja219x13kay3
```

---

All that said, we actually don't need to expose this to the build for the `nixpkgs-update` use case.  The bot is currently using `nix eval -f . pkgs.<attr>.drvAttrs.<query>`, which won't work with passthru, but we can easily update it to make a query like `nix eval -f . <attr>.<query>`, which *does* work with passthru, so I've updated the commit accordingly.

Reply:
Right, it just avoids yet another environment variable exposed to the builder as well as nix shell environments.